test/unit-tests/services/help-service.ts

```diff
@@ -434,5 +439,70 @@ and another one`
 				assert.isTrue(output.indexOf("index data is read") >= 0);
 			});
 		});
+
+		describe("extensions tests", () => {
+			const assertData = async (expectedEnumerateFilesInDirectorySyncCalledCounter: number, extensionsData?: IExtensionData[]): Promise<void> => {
+				const injector = createTestInjector({ isProjectTypeResult: false, isPlatformResult: true });
+				const $staticConfig = injector.resolve<Config.IStaticConfig>("staticConfig");
+				$staticConfig.MAN_PAGES_DIR = "man_pages_dir";
+				$staticConfig.HTML_PAGES_DIR = "html_pages_dir";
+				$staticConfig.CLIENT_NAME = "client name";
+				const extensionDocsDir = "extensionDocsDir";
+				let enumerateFilesInDirectorySyncCalledCounter = 0;
+				injector.register("fs", {
+					enumerateFilesInDirectorySync: (path: string) => {
+						enumerateFilesInDirectorySyncCalledCounter++;
+						if (path === extensionDocsDir) {
+							return [join(extensionDocsDir, "foo.md")];
+						}
+
+						return [];
+					},
+					readText: (pathToRead: string) => {
+						return pathToRead === join(extensionDocsDir, "foo.md") ? "bla end" : "";
+					}
+				});
+
+				const $extensibilityService = injector.resolve<IExtensibilityService>("extensibilityService");
+				extensionsData = extensionsData || [];
+				extensionsData.push({
+					extensionName: "extension3",
+					docs: extensionDocsDir,
+					version: "1.2.3",
+					pathToExtension: "extension3path"
+				});
+
+				$extensibilityService.getInstalledExtensionsData = (): IExtensionData[] => extensionsData;
+
+				const helpService = injector.resolve<IHelpService>("helpService");
+				await helpService.showCommandLineHelp("foo");
+				const output = injector.resolve("logger").output;
+				assert.isTrue(output.indexOf("bla end") >= 0);
+
+				assert.equal(enumerateFilesInDirectorySyncCalledCounter, expectedEnumerateFilesInDirectorySyncCalledCounter, "The enumerateFilesInDirectorySync method must be called exactly three times - one for CLI help, one for extension2 and one for extension3");
```

Comment:
This error message is incorrect because the number of called times differs and the extensions differ with each test